### PR TITLE
Signup page

### LINKED ISSUE #19: Sign Up Page
Implement Simple validation password should be hidden, emails should have "@" included.
City functionality.
Facebook, google

image

### PR DESCRIPTION
**Pull request by udenizdemirbilek**
#19 Draft commit for error checking
Signup Form is functional, creates a user in [firebase](https://console.firebase.google.com/project/capstone-turkey-giftify/authentication/users), 
image
stores the information in the [database](https://console.firebase.google.com/project/capstone-turkey-giftify/firestore/data/~2Fusers)
image

Also google sign-in is also working, please see first image.

Desktop 
image
Tablet
image
Mobile
image

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -1,33 +1,33 @@
-# This workflow will do a clean install of node dependencies, cache/restore them, build the source code and run tests across different versions of node
-# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
-
-name: Node.js CI
-
-on:
-  push:
-    branches: [ dev ]
```
Comment (halitbatur) [inline .github/workflows/node.js.yml]: there's a lot of files here that should not be changed please check end of lind error again and fix it

File: src/containers/SignUp/SignUp.jsx [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    const provider = new firebase.auth.GoogleAuthProvider();
+    firebase
+      .auth()
+      .signInWithPopup(provider)
+      .then((result) => {
+        /** @type {firebase.auth.OAuthCredential} */
+        const { credential } = result;
+        console.log(credential);
+
+        // This gives you a Google Access Token. You can use it to access the Google API.
```
Comment (halitbatur) [inline src/containers/SignUp/SignUp.jsx]: can we have this comment on top of line 79

File: src/containers/SignUp/SignUp.jsx [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+  // LogIn with Google
+  const logInGoogle = () => {
+    const provider = new firebase.auth.GoogleAuthProvider();
+    firebase
+      .auth()
+      .signInWithPopup(provider)
+      .then((result) => {
+        /** @type {firebase.auth.OAuthCredential} */
+        const { credential } = result;
+        console.log(credential);
```
Comment (halitbatur) [inline src/containers/SignUp/SignUp.jsx]: no console.logs please

File: src/containers/SignUp/SignUp.jsx [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+        // const token = credential.accessToken;
+        // The signed-in user info.
+        // const { user } = result;
+        // ...
+      })
+      .catch((error) => {
+        // Handle Errors here.
+        const errorCode = error.code;
+        const errorMessage = error.message;
+        console.log(errorCode, errorMessage);
```
Comment (halitbatur) [inline src/containers/SignUp/SignUp.jsx]: no console.logs

File: src/containers/SignUp/SignUp.jsx [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+      .then((result) => {
+        /** @type {firebase.auth.OAuthCredential} */
+        const { credential } = result;
+
+        // The signed-in user info.
+        const { user } = result;
+
+        // This gives you a Facebook Access Token. You can use it to access the Facebook API.
+        const { accessToken } = credential;
+        console.log(accessToken, user);
```
Comment (halitbatur) [inline src/containers/SignUp/SignUp.jsx]: no console.logs

File: src/containers/SignUp/SignUp.jsx [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+      })
+      .catch((error) => {
+        // Handle Errors here.
+        const errorCode = error.code;
+        const errorMessage = error.message;
+        // The email of the user's account used.
+        // const {email} = error;
+        // The firebase.auth.AuthCredential type that was used.
+        // const {credential} = error;
+        console.log(errorCode, errorMessage);
```
Comment (halitbatur) [inline src/containers/SignUp/SignUp.jsx]: no console.logs

**Review by halitbatur — CHANGES_REQUESTED** [review]

**Review by ceyhungulbas — CHANGES_REQUESTED** [review]
Hey Deniz! Eline saglik. Looks like its really looking good and functionality workin nice but I just found teeny issue about functionality.

resim

- As you see I signed up but my mail is not displaying on doc field, is there a way to add this info? Because we are going to use this information sending email between buyer and seller.
- For Re-Enter Password part: I wrote first aha123 then ehe123 then clicked sign up. But got any error message, just refreshing page. If we have a time, maybe we can add this thing too?

**Comment by udenizdemirbilek** [comment]
> 
> 
> Hey Deniz! Eline saglik. Looks like its really looking good and functionality workin nice but I just found teeny issue about functionality.
> 
> resim
> 
>     * As you see I signed up but my mail is not displaying on doc field, is there a way to add this info? Because we are going to use this information sending email between buyer and seller.
> 
>     * For Re-Enter Password part: I wrote first aha123 then ehe123 then clicked sign up. But got any error message, just refreshing page. If we have a time, maybe we can add this thing too?

Fixed both issues, also signing up with google and facebook should also create a database user entry with email, name and photoURL now.

Added a new rule to check two passwords to be the same and alerts the user if they are not.

**Review by ceyhungulbas — APPROVED** [review]
functionality is working very well. Lookin good! Eline saglik Deniz

File: src/containers/SignUp/SignUp.jsx [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+          <Form.Control
+            as="select"
+            name="city"
+            onChange={handleChange}
+            value={formData.city}
+            required
+          >
+            {cities.map((city, index) => (
+              // eslint-disable-next-line react/no-array-index-key
+              <option key={index}>{city}</option>
```
Comment (halitbatur) [inline src/containers/SignUp/SignUp.jsx:217]: lets not have the index as a key here maybe it can be the city

**Review by halitbatur — APPROVED** [review]

File: src/containers/SignUp/SignUp.jsx (hunk 7 above)
Comment (udenizdemirbilek) [reply src/containers/SignUp/SignUp.jsx:217]: Taken a note in the bug bash file!

**Pull request merged**